presto-main/src/main/java/com/facebook/presto/cost/CostComparator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+
+import javax.inject.Inject;
+
+import static com.facebook.presto.SystemSessionProperties.getQueryMaxMemory;
+
+public class CostComparator
+{
+    private final double cpuWeight;
+    private final double memoryWeight;
+    private final double networkWeight;
+
+    @Inject
+    public CostComparator(FeaturesConfig featuresConfig)
+    {
+        this(featuresConfig.getCpuCostWeight(), featuresConfig.getMemoryCostWeight(), featuresConfig.getNetworkCostWeight());
+    }
+
+    public CostComparator(double cpuWeight, double memoryWeight, double networkWeight)
+    {
+        this.cpuWeight = cpuWeight;
+        this.memoryWeight = memoryWeight;
+        this.networkWeight = networkWeight;
+    }
+
+    public int compare(Session session, PlanNodeCostEstimate left, PlanNodeCostEstimate right)
```

Comment:
Even if `CostComparator` uses session (currently it's only for the bogus formula `getQueryMaxMemory(session).toBytes()`), it would be way more convenient if it implemented `Comparator`. See teradata/presto#559 for example.

It could be as:

```
public Comparator<PlanNodeCostEstimate> {with,for}Session(Session session)
```